istioctl/pkg/install/pre-check.go

```diff
@@ -300,3 +319,186 @@ func (c *preCheckClient) checkMutatingWebhook() error {
+			targetNamespace := istioNamespace
+			targetRevision := opts.Revision
+			specific := c.Flags().Changed("istioNamespace") // is user asking about a specific Istio System ns or revision
+
+			// Check if we can install the IOP specified with -f
+			if len(fileNameFlags.ToOptions().Filenames) > 0 {
+				iop, err := getIOPFromFile(kubeConfigFlags, fileNameFlags.ToOptions())
+				if err != nil {
+					return err
+				}
+				// Currently we don't look at specific IOP options, just the namespace and Revision
+				targetNamespace = iop.GetNamespace()
+				targetRevision = iop.Spec.Revision
+				specific = true
+			}
+
+			cli, err := clientFactory(kubeConfigFlags)
+			if err != nil {
+				return err
+			}
+
+			installs, err := cli.getIstioInstalls()
+			if err == nil && len(installs) > 0 {
+				matched := false
+				for _, install := range installs {
+					if !specific || targetNamespace == install.namespace && targetRevision == install.revision {
+						c.Printf("Istio Revision %q already installed in namespace %q\n", install.revision, install.namespace)
+					}
+					if targetNamespace == install.namespace && targetRevision == install.revision {
+						matched = true
+					}
+				}
+				// The user has Istio, but wants to install a new revision
+				if !matched {
+					return installPreCheck(targetNamespace, kubeConfigFlags, c.OutOrStdout())
+				}
+				return nil
+			}
+
+			// No IstioOperator was found.  In 1.6.0 we fall back to checking for Istio namespace
+			nsExists, err := namespaceExists(targetNamespace, kubeConfigFlags)
+			if err != nil {
+				return err
+			}
+			if !nsExists {
+				return installPreCheck(targetNamespace, kubeConfigFlags, c.OutOrStdout())
+			}
+
+			// The Istio namespace does exist, but it wasn't installed by 1.6.0+ because no
+			// IstioOperator is there.
+			c.Printf("Istio already installed in namespace %q\n.  Skipping pre-check.", targetNamespace)
+			c.Printf("Use 'istioctl upgrade' to upgrade or 'istioctl install --set revision=<revision>' to install another control plane.")
+			return nil
+		},
+	}
+
+	flags := precheckCmd.PersistentFlags()
+	flags.StringVarP(&istioNamespace, "istioNamespace", "i", controller.IstioNamespace,
+		"Istio system namespace")
+	kubeConfigFlags.AddFlags(flags)
+	fileNameFlags.AddFlags(flags)
+	opts.AttachControlPlaneFlags(precheckCmd)
+	return precheckCmd
+}
+
+func findIstios(client dynamic.Interface) ([]istioInstall, error) {
+	retval := make([]istioInstall, 0)
+
+	// First, look for IstioOperator CRs left by 'istioctl manifest apply' or 'kubectl apply'
+	iops, err := allOperatorsInCluster(client)
+	if err != nil {
+		return retval, err
+	}
+	for _, iop := range iops {
+		retval = append(retval, istioInstall{
+			namespace: iop.Namespace,
+			revision:  iop.Spec.Revision,
+		})
+	}
+	return retval, nil
+}
+
+// nolint: lll
+func getIOPFromFile(restClientGetter resource.RESTClientGetter, options resource.FilenameOptions) (*operator_v1alpha1.IstioOperator, error) {
```

Comment:
If the cluster is not aware of the IOP crd, this method will fail like:
`unable to recognize "/Users/isisding/oti/goProjects/istio.io/istio/operator/data/profiles/iris.yaml": no matches for kind "IstioOperator" in version "install.istio.io/v1alpha1"`

Reply:
fixed